docs/sources/http_api/access_control.md

```diff
@@ -726,6 +726,218 @@ Content-Type: application/json; charset=UTF-8
 | 404  | Role not found.                                                      |
 | 500  | Unexpected error. Refer to body and/or server logs for more details. |
 
+## Create and remove team role assignments
+
+### List roles assigned to a team
+
+`GET /api/access-control/teams/:teamId/roles`
+
+Lists the roles that have been directly assigned to a given team.
+
+#### Required permissions
+
+| Action           | Scope                |
+| ---------------- | -------------------- |
+| teams.roles:list | teams:id:`<team ID>` |
+
+#### Example request
+
+```http
+GET /api/access-control/teams/1/roles
+Accept: application/json
+```
+
+#### Example response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=UTF-8
+
+[
+    {
+        "version": 4,
+        "uid": "j08ZBi-nk",
+        "name": "fixed:licensing:reader",
+        "displayName": "Licensing reader",
+        "description": "Read licensing information and licensing reports.",
+        "group": "Licenses",
+        "updated": "2022-02-03T14:19:50+01:00",
+        "created": "0001-01-01T00:00:00Z",
+        "global": false
+    }
+]
+```
+
+#### Status codes
+
+| Code | Description                                                          |
+| ---- | -------------------------------------------------------------------- |
+| 200  | Set of assigned roles is returned.                                   |
+| 403  | Access denied.                                                       |
+| 500  | Unexpected error. Refer to body and/or server logs for more details. |
+
+### Add a team role assignment
+
+`POST /api/access-control/teams/:teamId/roles`
+
+Assign a role to a specific team.
+
+For bulk updates consider [Set team role assignments]({{< ref "#set-team-role-assignments" >}}).
+
+#### Required permissions
+
+`permission:delegate` scope ensures that users can only assign roles which have same, or a subset of permissions which the user has.
+For example, if a user does not have required permissions for creating users, they won't be able to assign a role to a team which will allow to do that. This is done to prevent escalation of privileges.
```

Comment:
nit: I find this sentence slightly difficult to read. Maybe something like:
```suggestion
For example, if a user does not have the permissions required to create users, they won't be able to assign a role that contains these permissions. This is done to prevent escalation of privileges.
```